ascon/src/lib.rs

```diff
@@ -6,161 +9,422 @@
     html_favicon_url = "https://raw.githubusercontent.com/RustCrypto/meta/master/logo.svg"
 )]
 #![forbid(unsafe_code)]
-#![warn(
-    clippy::mod_module_files,
-    clippy::unwrap_used,
-    missing_docs,
-    rust_2018_idioms,
-    unused_lifetimes,
-    unused_qualifications
-)]
+#![warn(missing_docs)]
 
-#[cfg(feature = "alloc")]
-extern crate alloc;
+use core::mem::size_of;
+
+/// Produce mask for padding.
+#[inline(always)]
+pub const fn pad(n: usize) -> u64 {
+    0x80_u64 << (56 - 8 * n)
+}
 
-#[cfg(feature = "aead")]
-pub mod aead;
+/// Clear bytes from a 64 bit word.
+#[inline(always)]
+pub const fn clear(word: u64, n: usize) -> u64 {
+    word & (0x00ffffffffffffff >> (n * 8 - 8))
+}
```

Comment:
It looks like you define this in this crate but only use it in `ascon-aead`.

Are there other potential applications? If not, should it be moved to `ascon-aead`?

Reply:
Indeed, that's a left-over from the initial `ascon-core` / `ascon-aead` split. I'll move it to `ascon-aead`. `ascon-hash` doesn't use it.